units/initrd-battery-check.service.in

```diff
@@ -0,0 +1,20 @@
+#  SPDX-License-Identifier: LGPL-2.1-or-later
+#
+#  This file is part of systemd.
+#
+#  systemd is free software; you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License as published by
+#  the Free Software Foundation; either version 2.1 of the License, or
+#  (at your option) any later version.
+
+[Unit]
+Description=Check battery level during early boot
+DefaultDependencies=no
+After=systemd-udevd.service
+Before=initrd-switch-root.service
```

Comment:
can you also add ConditionPathExists=/etc/initrd-release here

this makes sure that the unit is only run if invoked in the initrd, and is otherwise skipped (i.e. when running on the host).

(background: /etc/initrd-release is how we expect initrds to be tagged)

Reply:
Also, please replace Before=initrd-switch-root.service by Before=local-fs-pre.target

"local-fs-pre.target" is a special target that must be reached before any host fs is mounted. this is useful here, because we really don't want to mount any fs writable if we are going to hard shutdown very soon.

Reply:
> can you also add ConditionPathExists=/etc/initrd-release here

Maybe better to use `AssertPathExists=` to keep things consistent with other `initrd-*` units

Reply:
> > can you also add ConditionPathExists=/etc/initrd-release here
> 
> Maybe better to use `AssertPathExists=` to keep things consistent with other `initrd-*` units

Okay sir.